syncd/SaiSwitch.cpp

```diff
@@ -39,6 +43,10 @@ SaiSwitch::SaiSwitch(
 
     GlobalSwitchId::setSwitchId(m_switch_rid);
 
+#ifdef MDIO_ACCESS_USE_NPU
+    MdioIpcServer::setSwitchId(m_switch_rid);
+#endif
+
```

Comment:
For multiple switches in gbsyncd, the code is fine as it will return without doing anything. For syncd, we don't have MDIO buses mapped to multiple NPU. If we are going to support MDIO buses mapped to more than single NPU, the entire MDIO IPC server code has to be changed. We are not addressing the issue this time.

Reply:
i would suggest to move this out of the Switch.cpp and uset this in Syncd.cpp, since eveery new switch this setSwitchId will be set and you dont want  that since you only want one switch, also you still call static api, and i propopsed to make this proper c++ object